processing/src/main/java/io/druid/query/groupby/GroupByQueryEngine.java

```diff
@@ -322,13 +430,28 @@ public RowIterator(GroupByQuery query, final Cursor cursor, ByteBuffer metricsBu
       dimensionSpecs = query.getDimensions();
       dimensions = Lists.newArrayListWithExpectedSize(dimensionSpecs.size());
       dimNames = Lists.newArrayListWithExpectedSize(dimensionSpecs.size());
+      originalDimNames = Lists.newArrayListWithExpectedSize(dimensionSpecs.size());
+      dimHandlers = Lists.newArrayListWithExpectedSize(dimensionSpecs.size());
+      dimCapabilities = Lists.newArrayListWithExpectedSize(dimensionSpecs.size());
```

Comment:
Given the sheer number of lists of `dimensionSpecs.size()` length all holding information about the dims, I gotta wonder if there shouldn't be a class created that has all of these things as fields and just a single list of instances of that class?